src/bag3_testbenches/measurement/mos/sp.py

```diff
@@ -76,29 +90,51 @@ async def async_meas_pvt(self, name: str, sim_dir: Path, sim_db: SimulationDB, d
                           dict(pin='dbias', nin='d', type='dcfeed', value=''),
                           ])
 
-        # set vgs and vds sweep info
-        vds_val = self.specs['vds_val']
-        vgs_val = self.specs['vgs_val']
+        is_mc = 'monte_carlo_params' in self.specs['tbm_specs']
+        is_nmos: bool = self.specs['is_nmos']
 
         tbm_specs = dict(
             **self.specs['tbm_specs'],
             load_list=load_list,
             sim_envs=[pvt],
-            swp_info=[('vds', dict(type='LIST', values=vds_val)),
-                      ('vgs', dict(type='LIST', values=vgs_val))],
             dut_pins=dut.pin_names,
             param_type='Y',
             ports=['PORTG', 'PORTD'],
         )
+
+        if not is_mc:
+            vgs_sweep = self.specs['vgs_sweep']
+            if is_nmos:
+                vgs_start, vgs_stop = vgs_sweep['vgs_min'], vgs_sweep['vgs_max']
+            else:
+                vgs_start, vgs_stop = -vgs_sweep['vgs_max'], -vgs_sweep['vgs_min']
+            tbm_specs['swp_info'] = [('vgs', dict(type='LINEAR', start=vgs_start, stop=vgs_stop, num=vgs_sweep['vgs_num']))]
+
         tbm = cast(SPTB, self.make_tbm(SPTB, tbm_specs))
+
+        # set vds
+        vds_val = self.specs['vds_val']
+        tbm.sim_params['vds'] = vds_val if is_nmos else -vds_val
+
+        # set vgs
+        if is_mc:
+            vgs_val = self.specs['vgs_val']
+            tbm.sim_params['vgs'] = vgs_val if is_nmos else -vgs_val
+
         sim_results = await sim_db.async_simulate_tbm_obj(name, sim_dir, dut, tbm, {'dut_conns': {'b': 's', 'd': 'd',
                                                                                                   'g': 'g', 's': 's'}},
                                                           harnesses=harnesses)
-        results = calc_ft_fmax(sim_results.data)
+        meas_type: str = self.specs['meas_type']
```

Comment:
No. Computing both `ft_fmax` and `gm_ro` simultaneously is computationally inefficient.
For `ft_fmax`, you need the frequency sweep to be in 10's of GHz to THz range to get the zero crossing of current gain or unilateral gain.
For `gm_ro`, you need the frequency sweep to be in KHz or MHz to single digit GHz range to avoid effects of high frequency parasitics.
So you want two very different non-overlapping frequency ranges for the s parameter simulation for the two cases, and it is extremely inefficient to have a wide frequency sweep that covers both. Technically `gm_ro` can also be done from DC sweeps like `ids` vs `vgs` and `ids` vs `vds`.